packages/core-data/src/test/resolvers.js

```diff
@@ -32,68 +36,100 @@ describe( 'getEntityRecord', () => {
 			baseURLParams: { context: 'edit' },
 		},
 	];
+	beforeEach( async () => {
+		triggerFetch.mockReset();
+		jest.useFakeTimers();
+	} );
 
 	it( 'yields with requested post type', async () => {
-		const fulfillment = getEntityRecord( 'root', 'postType', 'post' );
-		// Trigger generator
-		fulfillment.next();
-		// Provide entities and acquire lock
-		expect( fulfillment.next( ENTITIES ).value.type ).toEqual(
-			'@@data/DISPATCH'
-		);
-		// trigger apiFetch
-		const { value: apiFetchAction } = fulfillment.next();
-		expect( apiFetchAction.request ).toEqual( {
+		const dispatch = Object.assign( jest.fn(), {
+			receiveEntityRecords: jest.fn(),
+			__unstableAcquireStoreLock: jest.fn(),
+			__unstableReleaseStoreLock: jest.fn(),
+		} );
+		// Provide entities
+		dispatch.mockReturnValueOnce( ENTITIES );
+
+		// Provide response
+		triggerFetch.mockImplementation( () => POST_TYPE );
```

Comment:
Returning a promise from the mock implementation (there is a `triggerFetch.mockResolvedValue( POST_TYPE )` helper for that) is closer to what the unmocked `triggerFetch` does. Although `await` doesn't really care about the difference.